extensions/dacpac/src/wizard/pages/deployPlanPage.ts

```diff
@@ -103,6 +106,13 @@ export class DeployPlanPage extends DacFxConfigPage {
 				value: loc.dataLossTextWithCount(result.dataLossAlerts.size)
 			});
 			this.dataLossCheckbox.enabled = true;
+			this.model.potentialDataLoss = true;
+
+			// This will send telemetry event if potential data loss exists
+			TelemetryReporter.createActionEvent(TelemetryViews.DeployPlanPage, 'potentialDataLoss')
```

Comment:
This isn't really an action. What information are you looking to capture here? Do you actually care about seeing how often people see this message (which is basically what this is doing if I'm reading this correctly)? Or would it be better to include this kind of information in the event when the deployment is actually started?

Reply:
PotentialDataLoss property will capture if there is any data loss during pre deploy occurs, sending telemetry at this place will let us know if there is any data loss before **cancelling the wizard** without proceeding to generate Script or Deploy services.

Reply:
Pardon if I get this wrong (not really familiar with this stuff) - but isn't this just saying there's _potential_ data loss? At which step are the operations actually deployed that would result in the data loss?

@kisantia @udeeshagautam

Reply:
yup this just means there's potential data loss and the data loss would happen during deployment. I think this telemetry event is meant to see if people cancel out of the wizard or go back and change the target database if they see there's potential data loss

Reply:
You're not really going to get that information from this directly though since it'll be fired anytime there's potential data loss but that isn't going to correlated to whether they then continued on or canceled out. 

Ideally what should be happening is that the "close" events (either through attempted deployment or cancellation) should have additional properties such as `potentialDataLoss` that capture this - so when the wizard is cancelled we can see what page it was on and whether there was potential data loss that might have caused the user to cancel out.

I'd take a step back though and try to understand why you're capturing this data point. Are you just looking to find out how many people are quitting here because they saw the warning and didn't want that? That's going to be hard to know for sure - how do you know they didn't cancel for some other reason?

Reply:
Yes, we cannot guarantee if people quits only by seeing the data loss, there could be other reasons too. But if the data loss table pops up, this event will be recorded for potential data loss. Later, the available options will be Generate Deploy Script, Deploy, and cancel, for these the telemetry info would be covered at their respective methods, such as GenerateDeployScript() and Deploy()[including the potential Data loss, if these events fired which means user had proceeded with data loss] and for cancel/esc the hide()[at wizard level with what page they are cancelling the wizard]. Using these telemetry events order, at least we can conclude that user has closed after the data loss info displayed.

Reply:
Currently trying to order events like that isn't something we do and getting that set up is not going to be easy. 

The best way would be to attach the properties you want to the close events - that way you don't have to do any extra post processing of the events and are guaranteed to have the necessary data each time the wizard is closed.

Reply:
@Charles-Gagnon, for the additional properties(Ex: is **PotentialDataLoss** exists) to attach to cancel event. since, these props will build dynamically based on user actions to a particular event(let say, generateDeployReport), cancelling after it should attach them to the cancel event.

Thinking of adding a new parameter to the wizard interface and when the required props available with the generatedeployscript action, sending them with a callback to the wizard and updating the property and eventually using it at cancel event. would it be a suggestible approach? Any suggestions? Thanks :)

Reply:
That's going to be a bit tricky to do. For this case I'd suggest just doing your own cancel event by hooking into the onClick handler of the wizard cancel button on the extension side.